src/chainstate/coordinator/mod.rs

```diff
@@ -265,6 +265,14 @@ impl RewardSetProvider for OnChainRewardSetProvider {
         sortdb: &SortitionDB,
         block_id: &StacksBlockId,
     ) -> Result<RewardSet, Error> {
+        let cur_epoch = SortitionDB::get_stacks_epoch(sortdb.conn(), current_burn_height)?.expect(
+            &format!("FATAL: no epoch for burn height {}", current_burn_height),
+        );
+        if cur_epoch.epoch_id >= StacksEpochId::Epoch22 {
+            info!("PoX reward cycle defaulting to burn in Epoch 2.2");
+            return Ok(RewardSet::empty());
```

Comment:
You might want to add a new integration test for the affirmation map system.  All it needs to do is execute an epoch 2.2 transition and verify that a bootstrapping node (or a node that's missing anchor blocks) will be able to synchronize with a node that isn't missing any data.  This would verify that declaring that the reward set is empty does not disrupt the logic that identifies whether or not a PoX anchor block was chosen.

Reply:
Are there existing integration tests that do something similar to this? That'd be a helpful starting place for writing such a test.

Reply:
Yeah -- have a look at `test_pox_reorg_one_flap`.  Just have epoch 2.2 transition somewhere in the "middle" of the test and add on a few additional reward cycles.

Reply:
Should the epoch 2.2 transition occur before the hidden cycle?

Reply:
Added in [d7827f7](https://github.com/stacks-network/stacks-blockchain/pull/3677/commits/d7827f7cfd20346b67055a3330d4dfbdfa8a0c07)